# 
# backport SVN commit 1028896 by winterz:
# 
# properly handle recurring events when checking if an event has already occurred during the day.
# 

Index: kontact/plugins/korganizer/summaryeventinfo.cpp
===================================================================
--- kontact/plugins/korganizer/summaryeventinfo.cpp     (révision 1031052)
+++ kontact/plugins/korganizer/summaryeventinfo.cpp     (révision 1031053)
@@ -171,8 +171,18 @@
       continue;
     }
     // If the event is already over, then it isn't upcoming. so don't print it.
-    if ( !ev->allDay() && currentDateTime > ev->dtEnd() ) {
-      continue;
+    if ( !ev->allDay() ) {
+      if ( ev->recurs() ) {
+        KDateTime kdt( date, QTime( 0, 0, 0 ), KSystemTimeZones::local() );
+        kdt = kdt.addSecs( -1 );
+        if ( currentDateTime > ev->recurrence()->getNextDateTime( kdt ) ) {
+          continue;
+        }
+      } else {
+        if ( currentDateTime > ev->dtEnd() ) {
+          continue;
+        }
+      }
     }

     SummaryEventInfo *summaryEvent = new SummaryEventInfo();
